### DISCUSSION TITLE
MVN build fails

### DISCUSSION THREAD

**Discussion by arlange** [eclipse-sirius/sirius-components#2993] [Q&A] [answered]
Hi,

I back with a question for the maven build. After successfully completing the frontend build (and copying the dist folder) I ran `mvn clean install -DskipTests -f packages/pom.xml`. 
Result:
![grafik](https://github.com/eclipse-sirius/sirius-web/assets/118437561/c4cec46b-4e9a-45a2-8965-f3002858f9fc)

Hope you can help.

Regards

**Comment by AxelRICHARD** [comment]
May be you should add this to your `$HOME/.m2/settings.xml` file:
```
<server>
      <id>github-flow</id>
      <username>$YOUR_GITHUB_USERNAME</username>
      <password>$YOUR_GITHUB_ACCESS_TOKEN</password>
</server>
```
It is required by https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-sample-application/pom.xml#L50
In the same manner, you might need :
```
<server>
      <id>github-bpmn</id>
      <username>$YOUR_GITHUB_USERNAME</username>
      <password>$YOUR_GITHUB_ACCESS_TOKEN</password>
</server>
```
It is required by https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-sample-application/pom.xml#L54

Regards,

**Reply by arlange** [reply]
Yeah I did that, but it did not solve the issue. 
Before the build fails I get a warning:

[INFO] ---------< org.eclipse.sirius:sirius-components-flow-starter >----------
[INFO] Building sirius-components-flow-starter 2023.12.11              [97/110]
[INFO] --------------------------------[ jar ]---------------------------------

[WARNING] The POM for fr.obeo.dsl.designer.sample.flow:fr.obeo.dsl.designer.sample.flow:jar:1.0.11-SNAPSHOT is missing, no dependency information available

[WARNING] The POM for fr.obeo.dsl.designer.sample.flow:fr.obeo.dsl.designer.sample.flow.edit:jar:1.0.11-SNAPSHOT is missing, no dependency information available

**Reply by AxelRICHARD** [reply]
Could you verify the content of your `$HOME/.m2/repository/fr/obeo/dsl/designer/sample/flow/fr.obeo.dsl.designer.sample.flow` please? If some of the files (jar, pom...) are missing may be there was an error while maven tried to download this artifact.
If so, you can try to delete the `$HOME/.m2/repository/fr/obeo/dsl/designer/sample/flow/fr.obeo.dsl.designer.sample.flow` folder (and the same for other dependency missing) and launch the backend build again.

**Reply by arlange** [reply]
The path does not exist. `$HOME/.m2/repository` does not contain a folder 'fr'.

**Reply by AxelRICHARD** [reply]
Could you try to add the following:
```
<repositories>
      <repository>
                  <id>github-flow</id>
                  <url>https://maven.pkg.github.com/ObeoNetwork/Flow-Designer</url>
      </repository>
<repositories>
```
into https://github.com/eclipse-sirius/sirius-web/blob/master/packages/starters/backend/sirius-components-flow-starter/pom.xml#L42 and retry to build the backend please?

**Reply by arlange** [reply, marked as answer]
That worked, thanks. 
I added: 

`<repositories>
      <repository>
                  <id>github-flow</id>
                  <url>https://maven.pkg.github.com/ObeoNetwork/Flow-Designer</url>
      </repository>
</repositories>`

**Reply by AxelRICHARD** [reply]
A [fix](https://github.com/eclipse-sirius/sirius-web/commit/5384f8e06f27476ad5cc4b31bf074641553f5695) has been merged in Sirius Web
